File: LanguAI.Backend/LanguAI.Backend/Core/Models/Image.cs
using System.ComponentModel.DataAnnotations;

namespace LanguAI.Backend.Core.Models;

public class Image
{
    [Key]
    [Required]
    public int Id { get; set; }

    [Required]
    public string Name { get; set; }

    public DateTime? Uploaded { get; set; } = DateTime.UtcNow;

    public bool IsDeleted { get; set; } = false;

    [Required]
    public string Type { get; set; }

    public virtual User User { get; set; }

    public virtual Post Post { get; set; }
}

File: LanguAI.Frontend/src/api/models/image.ts
/* tslint:disable */
/* eslint-disable */
import { Post } from '../models/post';
import { User } from '../models/user';
export interface Image {
  id: number;
  isDeleted?: boolean;
  name: string;
  post?: Post;
  type: string;
  uploaded?: string | null;
  user?: User;
}
